dffml/source/source.py

```diff
@@ -43,6 +76,35 @@ def __init__(self, parent: "BaseSource") -> None:
     async def record(self, key: str):
         """
         Get a record from the source or add it if it doesn't exist
+
+        Examples
+        --------
+        >>> async def main():
+        ...     async with ExampleSource(records=[Record("example")]) as testSource:
+        ...         # Open, update, and close
+        ...         async with testSource() as sourceContext:
+        ...             # Let's check out our records before calling `record` and `update`.
+        ...             ITER = sourceContext.records()
+        ...             async for ITEM in ITER:
+        ...                 print(ITEM)
+        ...             one = await sourceContext.record("one")
+        ...             # ... update one ...
+        ...             await sourceContext.update(one)
+        ...             # Let's check out our records after calling `record` and `update`.
+        ...             ITER = sourceContext.records()
+        ...             async for ITEM in ITER:
+        ...                 print(ITEM)
```

Comment:
```suggestion
        ...                 print(record.export())
```

As you can see record printing has become odd. We should probably avoid it and print the exported version.

Reply:
Yeah, thanks. I've added data to `example` and also added another record using ctx.record since it can also add new records if they dont exist previously. I'm leaving it without data to show that it simply adds a record without any data.

Reply:
Also, do we want "extra" to not show as empty?

Reply:
`extra` is meant as an area where sources can store other information they might need on a record to `update` it properly. It's not relevant here, no need to show any data in it.